src/components/dialogs/network-modifications/battery/modification/battery-modification-form.js

```diff
@@ -0,0 +1,134 @@
+import {
+    ActivePowerAdornment,
+    filledTextField,
+    gridItem,
+    GridSection,
+    ReactivePowerAdornment,
+} from '../../../dialogUtils';
+import Grid from '@mui/material/Grid';
+import React from 'react';
+import FloatInput from 'components/utils/rhf-inputs/float-input';
+import ReactiveLimitsForm from '../../generator/reactive-limits/reactive-limits-form';
+import { TextField } from '@mui/material';
+import FrequencyRegulation from '../../generator/set-points/frequency-regulation';
+
+const BatteryModificationForm = ({
+    batteryToModify,
+    updatePreviousReactiveCapabilityCurveTable,
+    equipmentId,
+}) => {
+    const batteryIdField = (
+        <TextField
+            size="small"
+            fullWidth
+            label={'ID'}
+            value={equipmentId}
+            InputProps={{
+                readOnly: true,
+            }}
+            disabled
+            {...filledTextField}
+        />
+    );
+
+    const batteryNameField = (
+        <TextInput
+            name={EQUIPMENT_NAME}
+            label={'Name'}
+            formProps={filledTextField}
+            previousValue={batteryToModify?.name}
+            clearable={true}
+        />
+    );
+
+    const maximumActivePowerField = (
+        <FloatInput
+            name={MAXIMUM_ACTIVE_POWER}
+            label={'MaximumActivePowerText'}
+            adornment={ActivePowerAdornment}
+            previousValue={batteryToModify?.maxP}
+            clearable={true}
+        />
+    );
+
+    const minimumActivePowerField = (
+        <FloatInput
+            name={MINIMUM_ACTIVE_POWER}
+            label={'MinimumActivePowerText'}
+            adornment={ActivePowerAdornment}
+            previousValue={batteryToModify?.minP}
+            clearable={true}
+        />
+    );
+
+    const activePowerSetPointField = (
+        <FloatInput
+            name={ACTIVE_POWER_SET_POINT}
+            label={'ActivePowerText'}
+            adornment={ActivePowerAdornment}
+            previousValue={batteryToModify?.targetP}
+            clearable={true}
+        />
+    );
+
+    const reactivePowerSetPointField = (
+        <FloatInput
+            name={REACTIVE_POWER_SET_POINT}
+            label={'ReactivePowerText'}
+            adornment={ReactivePowerAdornment}
+            previousValue={batteryToModify?.targetQ}
+            clearable={true}
+        />
+    );
+
+    return (
+        <>
+            <Grid container spacing={2}>
+                {gridItem(batteryIdField, 4)}
+                {gridItem(batteryNameField, 4)}
+            </Grid>
+            {/* Limits part */}
+            <GridSection title="Limits" />
+            <Grid container spacing={2}>
+                {gridItem(minimumActivePowerField, 4)}
+                {gridItem(maximumActivePowerField, 4)}
+            </Grid>
+            {/* Reactive limits part */}
+            <ReactiveLimitsForm
+                batteryInfos={batteryToModify}
```

Comment:
the component "ReactiveLimitsForm" has no props named "batteryInfos", he has "generatorToModify"
to make this more flexible:
rename props in component "generatorToModify" --> "equipmentToModify"
change batteryInfos={batteryToModify} with equipmentToModify={batteryToModify}